crates/iceberg/src/scan.rs

```diff
@@ -163,24 +164,32 @@ impl TableScan {
 
         Ok(iter(file_scan_tasks).boxed())
     }
+
+    pub async fn execute(
```

Comment:
```suggestion
    pub async fn to_arrow(
```

How about making things more explicit ?